src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,168 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    private static class Node<T> {
+        private Node<T> prev;
+        private Node<T> next;
+        private T value;
+
+        Node(Node<T> prev, T value, Node<T> next) {
+            this.value = value;
+            this.prev = prev;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(last, value, null);
+        if (last == null) {
+            first = newNode;
+        } else {
+            last.next = newNode;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkPositionIndex(index);
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> current = node(index);
+            Node<T> newNode = new Node<>(current.prev, value, current);
+            if (current.prev == null) {
+                first = newNode;
+            } else {
+                current.prev.next = newNode;
+            }
+            current.prev = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkElementIndex(index);
+        return node(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkElementIndex(index);
+        Node<T> node = node(index);
+        T oldValue = node.value;
+        node.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkElementIndex(index);
+        Node<T> node = node(index);
+        T value = node.value;
+
+        if (node.prev == null) {
+            first = node.next;
+        } else {
+            node.prev.next = node.next;
+        }
+
+        if (node.next == null) {
+            last = node.prev;
+        } else {
+            node.next.prev = node.prev;
+        }
+
+        size--;
+        return value;
     }
 
     @Override
     public boolean remove(T object) {
+        if (object == null) {
+            for (Node<T> current = first; current != null; current = current.next) {
+                if (current.value == null) {
+                    removeNode(current);
+                    return true;
+                }
+            }
+        } else {
+            for (Node<T> current = first; current != null; current = current.next) {
+                if (object.equals(current.value)) {
+                    removeNode(current);
+                    return true;
+                }
+            }
+        }
         return false;
```

Comment:
In the `remove` method, if the object is not found in the list, the method should ensure that the list remains unchanged. Currently, it returns false correctly, but it's good practice to explicitly ensure no modifications are made if the element is not found.